lib/Net/DRI/Protocol/EPP/Extensions/GracePeriod.pm

```diff
@@ -95,11 +95,12 @@ sub info_parse
  my $infdata=$mes->get_extension($ns,'infData');
  return unless defined $infdata;
 
- my $cs=$rinfo->{domain}->{$oname}->{status}; ## a Net::DRI::Protocol::EPP::Core::Status object
+ my $cs;
+ $cs = $rinfo->{domain}->{$oname}->{status}; ## a Net::DRI::Protocol::EPP::Core::Status object
```

Comment:
What I was suggesting was to add `$cs = Net::DRI::Protocol::EPP::Core::Status->new() unless $cs` because even with this change `$cs->add()` will fail since $cs won't be a status object...